code/modules/organs/subtypes/augment.dm

```diff
@@ -291,4 +291,45 @@
 
 /obj/item/organ/internal/augment/fuel_cell
 	name = "integrated fuel cell"
-	organ_tag = BP_AUG_FUEL_CELL
\ No newline at end of file
+	organ_tag = BP_AUG_FUEL_CELL
+
+// Geeves!
+/obj/item/organ/internal/augment/language
+	name = "language processor"
+	desc = "An augment installed into the head that interfaces with the user's neural interface, intercepting and assisting language faculties."
+	organ_tag = BP_AUG_LANGUAGE
+	parent_organ = BP_HEAD
+	var/list/augment_languages = list() // a list of languages that this augment will add. add your language to this
+	var/list/added_languages = list() // a list of languages that get added when it's installed. used to remove languages later. don't touch this.
+
+/obj/item/organ/internal/augment/language/replaced(var/mob/living/carbon/human/target, obj/item/organ/external/affected)
+	. = ..()
+	for(var/language in augment_languages)
+		if(!(language in target.languages))
+			target.add_language(language)
+			added_languages += language
```

Comment:
you don't need two lists here, just remove the language from the augment_languages list if it's already in the mob

Reply:
augment_languages is the baseline. it should not be edited, because otherwise it'll go all fucky if you take it out and put it in someone else. i've optimized the list usage in the removed proc, though.